mlrun/serving/server.py

```diff
@@ -102,27 +103,33 @@ def set_error_stream(self, error_stream):
     def _get_db(self):
         return mlrun.get_run_db(secrets=self._secrets)
 
-    def init(self, context, namespace, resource_cache=None, logger=None):
+    def init(
+        self, context, namespace, resource_cache: ResourceCache = None, logger=None
+    ):
         """for internal use, initialize all states (recursively)"""
 
         if self.error_stream:
             self._error_stream_object = OutputStream(self.error_stream)
-        self.resource_cache = resource_cache
+        self.resource_cache = resource_cache or ResourceCache()
         context = GraphContext(server=self, nuclio_context=context, logger=logger)
 
         context.stream = _StreamContext(self.parameters, self.function_uri)
         context.current_function = self._current_function
+        context.get_store_resource = self.resource_cache.resource_getter(
+            self._get_db(), self._secrets
+        )
+        context.get_table = self.resource_cache.get_table
```

Comment:
I'm a bit scared of this cache, let's talk about it offline

Reply:
u are reading too much into it, it has specific use, not generic, its for initializing the storey data pipeline, the pipeline reads the resources from the context object and resources are added to it before the pipeline/graph starts, not on every call.